interfaces/builtin/ros_opt_data.go

```diff
@@ -0,0 +1,65 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const rosOptDataSummary = `Allows read-only access to the static data such as xacro,yaml,urdf,stl,... in the standard /opt/ros folder`
+
+const rosOptDataBaseDeclarationSlots = `
+  ros-opt-data:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: false
```

Comment:
I missed this, we what's the intention here? that any snap can access this without asking for permission? in that case we usually omit this, if the intention is that snap ask for autoconnection then this should be true

Reply:
I explicitly set the value as I did not know the usual behavior, but if the coding style is to omit if true and it goes true by default, I will remove this line.

Reply:
sorry, I'm confused, you set it to false, so not sure why you mention true in your sentence. Should any snap installed on a system get access to these files? there no security consequences to that? I suppose so, but want to double check.

Reply:
sorry, my intention is for snaps to get autoconnected automatically, so this should be false because this is **deny**-auto-connection. My sentence above was written too quickly, if the default behavior is to get autoconnected automatically == true  -> I can omit this and remove the line. Already pushed without it.